docs/src/literate/helmholtz.jl

```diff
@@ -0,0 +1,170 @@
+    global_dofs = zeros(Int, ndofs_per_cell(dh))
+
+    fe = zeros(n_basefuncs) # Local force vector
+    Ke = zeros(n_basefuncs, n_basefuncs) # Local stiffness mastrix
+
+    @inbounds for (cellcount, cell) in enumerate(CellIterator(dh))
+        fill!(Ke, 0)
+        fill!(fe, 0)
+        coords = getcoordinates(cell)
+
+        reinit!(cellvalues, cell)
+        # First we derive the non boundary part of the variation problem from the destined solution `u_ana`
+        # ```math
+        # \int \nabla δu \cdot \nabla u d\Omega
+        # + \int δu \cdot u d\Omega
+        # - \int δu \cdot f d\Omega
+        # ```
+        #+
+
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+            coords_qp = spatial_coordinate(cellvalues, q_point, coords)
+            f_true = LinearAlgebra.tr(hessian(u_ana, coords_qp)) + u_ana(coords_qp)
+            for i in 1:n_basefuncs
+                δu = shape_value(cellvalues, q_point, i)
+                ∇δu = shape_gradient(cellvalues, q_point, i)
+                fe[i] += (δu * f_true) * dΩ
+                for j in 1:n_basefuncs
+                    u = shape_value(cellvalues, q_point, j)
+                    ∇u = shape_gradient(cellvalues, q_point, j)
+                    Ke[i, j] += (∇δu ⋅ ∇u + δu * u) * dΩ
+                end
+            end
+        end
+
+        # Now we manually add the von Neumann boundary terms
+        # ```math
+        # \int δu \cdot (n \cdot \nabla u - g_2) d\Gamma_2
+        # ```
+        #+
+        for face in 1:nfaces(cell)
+            if onboundary(cell, face) && 
+                   ((cellcount, face) ∈ getfaceset(grid, "left") || 
+                    (cellcount, face) ∈ getfaceset(grid, "bottom"))
+                reinit!(facevalues, cell, face)
+                for q_point in 1:getnquadpoints(facevalues)
+                    coords_qp = spatial_coordinate(facevalues, q_point, coords)
+                    n = getnormal(facevalues, q_point)
+                    g = gradient(u_ana, coords_qp) ⋅ n
+                    dΓ = getdetJdV(facevalues, q_point)
+                    for i in 1:n_basefuncs
+                        δu = shape_value(facevalues, q_point, i)
+                        fe[i] += (δu * g) * dΓ
+                    end
+                end
+            end
+        end
+   
+        celldofs!(global_dofs, cell)
+        assemble!(assembler, global_dofs, fe, Ke)
+    end
+    return K, f
+end;
+
+K, f = doassemble(cellvalues, facevalues, K, dh);
+apply!(K, f, dbcs)
+u = Symmetric(K) \ f;
+
+vtkfile = vtk_grid("helmholtz", dh)
+vtk_point_data(vtkfile, dh, u)
+vtk_save(vtkfile)
+using Test #src
+#src this test catches unexpected changes in the result over time
+#src it does not certify that the solution is any good
+Test.@test maximum(u) ≈ 0.05637592090022005 #src
+@assert u_ana(Vec{2}((-0.5, -0.5))) >= 1 #src
+@assert u_ana(Vec{2}((0.5, -0.5))) >= 1 #src
+@assert u_ana(Vec{2}((-0.5, 0.5))) >= 1 #src
```

Comment:
It is my understanding that since we use Lagrange elements, the entires of `u` are values of the solution at node points. But the maximum of `u_ana` seems to be slightly bigger then `1`? So this would be an indicator that our solution is garbage?

Reply:
Probably not complete garbage since the fields have at least similar shape :D. But maybe some factor wrong? Or something else. 

![bild](https://user-images.githubusercontent.com/1282691/68993747-b3dfa700-087b-11ea-82ea-dc1b612cefae.png)

![bild](https://user-images.githubusercontent.com/1282691/68993750-b6420100-087b-11ea-94c1-b4e886c8c00c.png)

Reply:
I will check if I spot an error.

Reply:
I think I fixed the errors, see here https://github.com/KristofferC/JuAFEM.jl/pull/261/commits/cf9f78c6c6ba8226be95a500b85de3b29b966f68

Reply:
How do you visualize the `.vtu` file? I tried using paraview, but it gives some errors and just shows a grey square.

Reply:
FWIW, i had the same problem and paraview worked well only after switching the linux session to open under the "xorg" mode instead of the default "wayland". I'm on a linux machine with Fedora 31.